### PR TITLE
Added Unicode scalar hex escapes (\u{1F913} for 🤓) to string syntax

### PR DESCRIPTION
**Pull request by cabo**

### CODE REVIEW

File: cbor-diag-parser.abnf [hunk 1]
```diff
@@ -75,12 +75,14 @@ escapable       = %s"b" ; BS backspace U+0008
                 / "\"   ; \ backslash (reverse solidus) U+005C
                 / (%s"u" hexchar) ;  uXXXX      U+XXXX
 
-hexchar         = non-surrogate
+hexchar         = "{" *"0" hexscalar "}"
```
Comment (hildjj) [inline cbor-diag-parser.abnf:78]: `*"0"` doesn't feel right.  Do you want `\u{00000000000001}` to be valid?  Maybe roll this into hexscalar for each alternative, bringing each up to a max of 6 characters.

File: cbor-diag-parser.abnf [hunk 2]
```diff
@@ -75,12 +75,14 @@ escapable       = %s"b" ; BS backspace U+0008
...
+                / non-surrogate
                 / (high-surrogate "\" %s"u" low-surrogate)
 non-surrogate   = ((DIGIT / "A"/"B"/"C" / "E"/"F") 3HEXDIG)
                 / ("D" ODIGIT 2HEXDIG )
 high-surrogate  = "D" ("8"/"9"/"A"/"B") 2HEXDIG
 low-surrogate   = "D" ("C"/"D"/"E"/"F") 2HEXDIG
+hexscalar       = "10" 4HEXDIG / HEXDIG1 4HEXDIG / non-surrogate / 0*3HEXDIG
```
Comment (hildjj) [inline cbor-diag-parser.abnf:85]: `\u{}` should not be valid, so this should be `1*3HEXDIG`, I think.

**Review by hildjj — CHANGES_REQUESTED** [review]

File: cbor-diag-parser.abnf (hunk 1 above)
Comment (cabo) [reply cbor-diag-parser.abnf:78]: Well, \U00001234 is quite common in other languages, so six is probably not making it too easy to convert that.
But this seems to be a stylistic thing anyway, I'd say; there is no specific artificial limit that can obviously agreed on.

Background is that I'm trying to keep this PEG-compatible, and there is only so much that can be achieved in the syntax.  I was trying to ensure surrogates trip the grammar, and that has priority I think.

File: cbor-diag-parser.abnf (hunk 2 above)
Comment (cabo) [reply cbor-diag-parser.abnf:85]: Another PEG compatibility thing.  The 0 is eaten by the greedy `*"0"`, and to properly disallow `\u{}` this needs two more lines.  Oh well.

**Review by hildjj — APPROVED** [review]

**Pull request merged**